packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -607,7 +607,7 @@ class _DefaultPub implements Pub {
         'permissions for the current user.'
       );
     }
-    return <String>[sdkPath, '__deprecated_pub', ...arguments];
+    return <String>[sdkPath, '--no-analytics', 'pub', ...arguments];
```

Comment:
you don't ever want analytics from these invocations? and is `__deprecated_pub` already not sending analytics?

Reply:
I think eventually we do want. But I wanted this PR to have minimal behavioral change. And we will probably have to revisit the analytics soon for the new pipeline anyway.

> is __deprecated_pub already not sending analytics?

No - that has been an outstanding issue.

Reply:
SGTM